runtime/account_keys_test.go

```diff
@@ -0,0 +1,937 @@
+			},
+			emitEvent: func(event cadence.Event) error {
+				events = append(events, event)
+				return nil
+			},
+			decodeArgument: func(b []byte, t cadence.Type) (value cadence.Value, err error) {
+				return json.Decode(b)
+			},
+		}
+
+		t.Run(tt.name, func(t *testing.T) {
+			args := make([][]byte, len(tt.args))
+			for i, arg := range tt.args {
+				var err error
+				args[i], err = json.Encode(arg)
+				if err != nil {
+					panic(fmt.Errorf("broken test: invalid argument: %w", err))
+				}
+			}
+
+			err := runtime.ExecuteTransaction(
+				Script{
+					Source:    []byte(tt.code),
+					Arguments: args,
+				},
+				Context{
+					Interface: runtimeInterface,
+					Location:  utils.TestLocation,
+				},
+			)
+			require.NoError(t, err)
+			assert.Len(t, events, tt.keyCount+1)
+			assert.Len(t, keys, tt.keyCount)
+			assert.Equal(t, tt.expected, keys)
+
+			assert.EqualValues(t, stdlib.AccountCreatedEventType.ID(), events[0].Type().ID())
+
+			for _, event := range events[1:] {
+				assert.EqualValues(t, stdlib.AccountKeyAddedEventType.ID(), event.Type().ID())
+			}
+		})
+	}
+}
+
+func TestAccountKeyCreation(t *testing.T) {
+
+	t.Parallel()
+
+	runtime := NewInterpreterRuntime()
+
+	script := []byte(`
+		pub fun main(): AccountKey {
+			let key = AccountKey(
+				PublicKey(
+					publicKey: "0102".decodeHex(),
+					signAlgo: "SignatureAlgorithmECDSA_P256"
+				),
+				hashAlgo: "HashAlgorithmSHA3_256",
+				weight: 1.7
+			)
+
+			return key
+      	}
+	`)
+
+	runtimeInterface := &testRuntimeInterface{}
+
+	_, err := runtime.ExecuteScript(
+		Script{
+			Source: script,
+		},
+		Context{
+			Interface: runtimeInterface,
+			Location:  utils.TestLocation,
+		},
+	)
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "cannot find variable in this scope: `AccountKey`")
+}
+
+func TestImportExportKeys(t *testing.T) {
+	t.Parallel()
+
+	runtime := NewInterpreterRuntime()
+
+	var tests = []accountKeyTestCase{
+		{
+			name: "AccountKey as transaction param",
+			code: `
+				pub fun main(key: AccountKey): PublicKey {
+					return key.publicKey
+				}
+			`,
+			args: []cadence.Value{accountKeyExportedValue(
+				0,
+				[]byte{1, 2, 3},
+				sema.SignatureAlgorithmECDSA_P256,
+				sema.HashAlgorithmSHA3_256,
+				"100.0",
+				true,
+			)},
+		},
+	}
+
+	for _, test := range tests {
+		storage := newTestAccountKeyStorage()
+		runtimeInterface := getRuntimeInterface(storage)
+
+		t.Run(test.name, func(t *testing.T) {
+			value, err := executeScript(test, runtime, runtimeInterface)
+			require.NoError(t, err)
+			require.NotNil(t, value)
+
+			expectedValue := publicKeyExportedValue(
+				[]byte{1, 2, 3},
+				sema.SignatureAlgorithmECDSA_P256,
+			)
+			assert.Equal(t, expectedValue, value)
+		})
+	}
+}
+
+func TestImportInvalidType(t *testing.T) {
+	t.Parallel()
+
+	runtime := NewInterpreterRuntime()
+
+	// encoded with an invalid type: 'N.PublicKey'
+	encodedArgs := []byte(`{
+		"type":"Struct",
+		"value":{
+			"id":"N.PublicKey",
```

Comment:
Is it invalid because of the unknown location prefix?

Reply:
yeah. 
Maybe this is not the best place for the test case though.

Reply:
Moved under json encoding/decoding test: f436f41